File: dev-docs/snippets/events/permission.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Fires when this client's permission level on a document changes.
export function permission(client: JsBaoClient) {
  // #region example
  client.on("permission", (payload) => {
    console.log(payload.documentId, "->", payload.permission);
  });
  // #endregion example
}

File: dev-docs/snippets/events/permission.swift
import JsBaoClient

// Fires when this client's permission level on a document changes.
func permission(client: JsBaoClient) {
  // #region example
  let sub = client.events.on(.permission) { (event: PermissionEvent) in
    print(event.documentId, "->", event.permission.rawValue)
  }
  // #endregion example
  _ = sub
}
